CatAnalyzer/plugins/ttbbLepJetsAnalyzer.cc

```diff
@@ -732,10 +788,44 @@ void ttbbLepJetsAnalyzer::analyze(const edm::Event& iEvent, const edm::EventSetu
     // DR 
     b_DRAddJets = genttbarConeCat->begin()->dRaddJets();
 
+    // adding b jet four-momentum having minimum deltaR
+    std::vector<math::XYZTLorentzVector> genbjets = genttbarConeCat->begin()->bJets();
+    tmp2->Fill(genbjets.size());
+    double mindR=999;
+    int index1=0, index2=0;
+    for(unsigned int i=0; i < genbjets.size(); ++i){
+      if( genbjets[i].Pt() < 20 || std::abs(genbjets[i].Eta()) > 2.5 ) continue;
+      for(unsigned int j=0; j < genbjets.size(); ++j){
+	if( genbjets[j].Pt() < 20 || std::abs(genbjets[j].Eta()) > 2.5 ) continue;
+	if( i == j ) continue;
+	double tmp = reco::deltaR(genbjets[i], genbjets[j]);
+	if( tmp < mindR ){
+	  mindR = tmp;
+	  index1 = i;
+	  index2 = j;
+	}
+      }
+    }
+
+    b_mindRbjet1_pt = genbjets[index1].Pt();
+    b_mindRbjet1_eta = genbjets[index1].Eta();
+    b_mindRbjet1_phi = genbjets[index1].Phi();
+    b_mindRbjet1_e = genbjets[index1].E();
+
+    b_mindRbjet2_pt = genbjets[index2].Pt();
+    b_mindRbjet2_eta = genbjets[index2].Eta();
+    b_mindRbjet2_phi = genbjets[index2].Phi();
+    b_mindRbjet2_e = genbjets[index2].E();
```

Comment:
could be very rare - this will be crashing if number of genbjets == 0. 

In addition, it can give wrong answer in some cases such as:
- there's only one genbjet -> still gives index1 = index2 = 0
- there's two genbjet but the 0th one fails acceptance -> gives index1 = 1, index2 = 0